compiler/noirc_frontend/src/monomorphization/mod.rs

```diff
@@ -385,7 +385,11 @@ impl<'interner> Monomorphizer<'interner> {
                 let rhs = self.expr(infix.rhs);
                 let operator = infix.operator.kind;
                 let location = self.interner.expr_location(&expr);
-
+                use crate::BinaryOpKind::*;
+                if matches!(operator, Less | LessEqual | Greater | GreaterEqual) {
+                    let lhs_type = self.interner.id_type(infix.lhs);
+                    assert!(!lhs_type.is_field(), "Comparisons are invalid on Field types");
+                }
```

Comment:
Was this leftover from debugging?

Reply:
No, it is referring to this comment:
The compiler will stop here with a better panic if fields are still used in a comparison, until the issue is properly fixed.

But by specifying the Ord trait as you suggest below, the error is properly reported now, so I will remove this.